### PR TITLE
CT-3076 Fixed issue of setting up deadline for non-standard type

### PR DESCRIPTION
**Pull request by ymao2**
## Description

Fixed the bug of pre-populating the deadline for non-standard complaint

right now the majority of feature tests for complaints are all for standard type by default,  once tackle other complaint type like ico and litigation, new test specs need to be added to cover those missing cases. 

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3076

### CODE REVIEW

File: app/form_models/offender_sar_complaint_case_form.rb [hunk 1]
```diff
@@ -77,7 +77,7 @@ def params_after_step_confirm_offender_sar(params)
   end
 
   def params_after_step_date_received(params)
-    if [nil, "standard", "ico"].include? params["complaint_type"]
+    if [nil, "standard"].include? object["complaint_type"]
```
Comment (willmcb) [inline app/form_models/offender_sar_complaint_case_form.rb:80]: What is `object` in this context? Could we give it a less generic name?

**Review by willmcb — APPROVED** [review]
One small comment otherwise LGTM 👍

**Comment by willmcb** [comment]
Should we add a ticket to the backlog for the new complaint types?

File: app/form_models/offender_sar_complaint_case_form.rb (hunk 1 above)
Comment (ymao2) [reply app/form_models/offender_sar_complaint_case_form.rb:80]: it is indeed confusing,  actually this module is included in `OffenderBaseDecorator`, the `object` is the way designed from Draper::Decorator (super super class of `OffenderBaseDecorator`),  it is designed as generic name by purpose I guess

Comment (ymao2) [reply app/form_models/offender_sar_complaint_case_form.rb:80]: sorry for not explaining the `object,` the object is the actual activerecord model instance.  e.g. 
for `OffenderBaseDecorator`,  the object would be offender_sar case

**Pull request merged**